src/tools/ilasm/ILAssembler/NameHelpers.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace ILAssembler
+{
+    internal static class NameHelpers
+    {
+        public static (string Namespace, string Name) SplitDottedNameToNamespaceAndName(string dottedName)
```

Comment:
What's the native ilasm method that this is trying to match? Is it `ns::FindSep` and friends?

If it is, it should skip an extra dot.